streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -213,6 +224,177 @@ public void testUpdate() throws IOException {
         assertEquals(new Long(30L + 1L), offsets.get(partition2));
     }
 
+    @Test
+    public void testUpdateWindowed() throws IOException {
+        final String changelogName = "test-application-windowed-store-changelog";
+
+        final TopicPartition topicPartition = new TopicPartition(changelogName, 1);
+        final List<TopicPartition> partitions = Utils.mkList(topicPartition);
+
+        consumer.assign(partitions);
+        consumer.commitSync(Utils.mkMap(Utils.mkEntry(topicPartition, new OffsetAndMetadata(0L))));
+
+        final Serializer<Windowed<Integer>> windowedChangelogKeySerializer = new Serializer<Windowed<Integer>>() {
+            final IntegerSerializer integerSerializer = new IntegerSerializer();
+
+            @Override
+            public void configure(final Map<String, ?> configs, final boolean isKey) {
+
+            }
+
+            @Override
+            public byte[] serialize(final String topic, final Windowed<Integer> data) {
+                final Bytes wrap = Bytes.wrap(integerSerializer.serialize(null, data.key()));
+                return WindowKeySchema.toStoreKeyBinary(new Windowed<>(wrap, data.window()), 1).get();
+            }
+
+            @Override
+            public void close() {
+
+            }
+        };
+
+        final MockRestoreConsumer<Windowed<Integer>, Long> restoreStateConsumer = new MockRestoreConsumer<>(windowedChangelogKeySerializer, new LongSerializer());
+        final StoreChangelogReader changelogReader = new StoreChangelogReader(
+            restoreStateConsumer,
+            Duration.ZERO,
+            stateRestoreListener,
+            new LogContext("standby-task-test ")
+        );
+
+        restoreStateConsumer.updatePartitions(changelogName, Utils.mkList(
+            new PartitionInfo(changelogName, 0, Node.noNode(), new Node[0], new Node[0])));
+        final InternalStreamsBuilder builder = new InternalStreamsBuilder(new InternalTopologyBuilder());
+        final String storeName = "windowed-store";
+        builder.stream(Collections.singleton("topic"), new ConsumedInternal<>()).groupByKey().windowedBy(TimeWindows.of(60_000).until(120_000)).count(Materialized.as(storeName));
+
+        final StreamsConfig config = createConfig(baseDir);
+        final InternalTopologyBuilder internalTopologyBuilder = InternalStreamsBuilderTest.internalTopologyBuilder(builder);
```

Comment:
Do we need this? We pass in a `new InternalTopologyBuilder` above. Maybe just keep a reference to it directly?